features/step_definitions/membership_application_steps.rb

```diff
@@ -1,6 +1,8 @@
 And(/^the following applications exist:$/) do |table|
   table.hashes.each do |hash|
-    attributes = hash.except('user_email', 'categories', 'company_name', 'company_number')
+
+    attributes = hash.except('user_email', 'categories', 'company_name',
+                             'company_number', 'legacy')
```

Comment:
I don't like that because 1) we may end up having other criteria the comprise a "legacy" application, and 2) (less importantly) a label like that is way too big for the limited space represented by a _Background_ section.

Re the first reason, today we even talked about having to differentiate - for production _and_ testing purposes, a user account which was created before we required the GDRP confirmation, and one created afterwards.

And, yes, "legacy" is ambiguous - so we have to make sure we have adequate comments in the factory or step methods or both.

I made these changes:

1. I changed the attribute name to "is_legacy", to be consistent with what was done for the user model (when we refactored the name attributes from ShfApplication to the User model).

2. I added (or moved) comments re what is meant by "legacy" app to the top of the cucumber ShfApplication steps file.

3. I added comments to the steps file for User.

Reply:
The comments are really helpful.  (I still don't like the word "legacy"  in there at all.  But I think we'll all survive. ;-) )